tests/MongoDB.Driver.Core.TestHelpers/CoreTestConfiguration.cs

```diff
@@ -107,14 +108,17 @@ public static ClusterBuilder ConfigureCluster(ClusterBuilder builder)
                 .ConfigureWithConnectionString(__connectionString.Value)
                 .ConfigureCluster(c => c.With(serverSelectionTimeout: TimeSpan.FromMilliseconds(int.Parse(serverSelectionTimeoutString))));
 
-            if (__connectionString.Value.Tls.HasValue && __connectionString.Value.Tls.Value)
+            if (__connectionString.Value.Tls.HasValue &&
+                __connectionString.Value.Tls.Value &&
+                __connectionString.Value.AuthMechanism != null &&
+                __connectionString.Value.AuthMechanism == MongoDBX509Authenticator.MechanismName)
             {
-                var certificateFilename = Environment.GetEnvironmentVariable("MONGO_SSL_CERT_FILE");
```

Comment:
I tweaked the name here because  think X.509 certificate may be the more technically correct name. https://en.wikipedia.org/wiki/X.509

Reply:
As long as nothing was depending on the original name...

Reply:
Good point! Nothing I could find with a global search, even taking into account files outside the sln